examples/README

```diff
@@ -7,7 +7,7 @@ Examples:
   producer_channel_example - Channel based producer
   producer_example - Function based producer
 
-  go-kafkacat - Channel based kafkacat Go clone
+  go-kafkacat - Channel based kafkacat Go clone, also provides unsecured SASL/OAUTHBEARER example
```

Comment:
Can we move this to its own minimal example?
I think something that simply lists the brokers in the cluster (using GetMetadata()) is sufficient to show-case the OAUTHBEARER use.

I've done the mistake of trying to add everything to the same example before, and it just turns out to be a bad example (looking at you, [rdkafka_example.c](https://github.com/edenhill/librdkafka/blob/master/examples/rdkafka_example.c)).

Reply:
Thanks for adding `consumer_test.go` and `producer_test.go` while I was away.